m3/src/test/java/com/uber/m3/tally/m3/thrift/TUdpClientTest.java

```diff
@@ -0,0 +1,72 @@
+package com.uber.m3.tally.m3.thrift;
+
+import org.apache.commons.codec.Charsets;
+import org.apache.thrift.transport.TTransportException;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.util.List;
+
+import static org.mockito.Mockito.*;
+
+public class TUdpClientTest {
+
+    @Test
+    public void testWritingAfterFlushingSequence() throws TTransportException, IOException {
+        DatagramSocket mockedSocket = mock(DatagramSocket.class);
+
+        TUdpClient client = createUdpClient(mockedSocket);
```

Comment:
is there a mechanism to move this to some kind of setup pattern?

Reply:
There's just one test for now, so not sure if worth extracting